elasticsearch_dsl/interfaces.py

```diff
@@ -0,0 +1,3283 @@
+    :arg query: (required) Terms you wish to find in the provided field.
+        The last term is used in a prefix query.
+    :arg boost: Floating point number used to decrease or increase the
+        relevance scores of the query. Boost values are relative to the
+        default value of 1.0. A boost value between 0 and 1.0 decreases
+        the relevance score. A value greater than 1.0 increases the
+        relevance score.
+    :arg _name: No documentation available.
+    """
+
+    analyzer: Union[str, "DefaultType"]
+    fuzziness: Union[str, int, "DefaultType"]
+    fuzzy_rewrite: Union[str, "DefaultType"]
+    fuzzy_transpositions: Union[bool, "DefaultType"]
+    max_expansions: Union[int, "DefaultType"]
+    minimum_should_match: Union[int, str, "DefaultType"]
+    operator: Union[Literal["and", "or"], "DefaultType"]
+    prefix_length: Union[int, "DefaultType"]
+    query: Union[str, "DefaultType"]
+    boost: Union[float, "DefaultType"]
+    _name: Union[str, "DefaultType"]
+
+    def __init__(
+        self,
+        *,
+        analyzer: Union[str, "DefaultType"] = DEFAULT,
+        fuzziness: Union[str, int, "DefaultType"] = DEFAULT,
+        fuzzy_rewrite: Union[str, "DefaultType"] = DEFAULT,
+        fuzzy_transpositions: Union[bool, "DefaultType"] = DEFAULT,
+        max_expansions: Union[int, "DefaultType"] = DEFAULT,
+        minimum_should_match: Union[int, str, "DefaultType"] = DEFAULT,
+        operator: Union[Literal["and", "or"], "DefaultType"] = DEFAULT,
+        prefix_length: Union[int, "DefaultType"] = DEFAULT,
+        query: Union[str, "DefaultType"] = DEFAULT,
+        boost: Union[float, "DefaultType"] = DEFAULT,
+        _name: Union[str, "DefaultType"] = DEFAULT,
+        **kwargs: Any,
+    ):
+        if analyzer != DEFAULT:
+            kwargs["analyzer"] = analyzer
+        if fuzziness != DEFAULT:
+            kwargs["fuzziness"] = fuzziness
+        if fuzzy_rewrite != DEFAULT:
+            kwargs["fuzzy_rewrite"] = fuzzy_rewrite
+        if fuzzy_transpositions != DEFAULT:
+            kwargs["fuzzy_transpositions"] = fuzzy_transpositions
+        if max_expansions != DEFAULT:
+            kwargs["max_expansions"] = max_expansions
+        if minimum_should_match != DEFAULT:
+            kwargs["minimum_should_match"] = minimum_should_match
+        if operator != DEFAULT:
+            kwargs["operator"] = operator
+        if prefix_length != DEFAULT:
+            kwargs["prefix_length"] = prefix_length
+        if query != DEFAULT:
+            kwargs["query"] = query
+        if boost != DEFAULT:
+            kwargs["boost"] = boost
+        if _name != DEFAULT:
+            kwargs["_name"] = _name
+        super().__init__(**kwargs)
+
+
+class MatchPhrasePrefixQuery(QueryBase):
+    """
+    :arg analyzer: Analyzer used to convert text in the query value into
+        tokens.
+    :arg max_expansions: Maximum number of terms to which the last
+        provided term of the query value will expand.
```

Comment:
I'm not sure I like the idea, because server defaults can be tied to individual versions of the stack, and that will make the library do strange things if you connect to a version that is not the one that was used to generate this code.

Reply:
I was talking about documentation only, not actually setting the default (it's a server default, not a client default).

Reply:
Ah, sorry! I can add the server default to the docstring, certainly.

Reply:
Defaults added!